include/swift/SIL/SILLinkage.h

```diff
@@ -374,7 +375,7 @@ inline SILLinkage effectiveLinkageForClassMember(SILLinkage linkage,
 inline bool
 fixmeWitnessHasLinkageThatNeedsToBePublic(SILDeclRef witness) {
   auto witnessLinkage = witness.getLinkage(ForDefinition);
-  return !hasPublicVisibility(witnessLinkage)
+  return !hasPublicVisibility(witnessLinkage, witness.getASTContext().SILOpts.EnableSerializePackage)
```

Comment:
IIUC, the symbol linkage of witnesses is independent on if package CMO is enabled